crates/hir_def/src/type_ref.rs

```diff
@@ -51,6 +51,18 @@ impl Rawness {
     }
 }
 
+#[derive(Clone, PartialEq, Eq, Hash, Debug)]
+pub struct TraitRef {
+    pub path: Path,
+}
+
+impl TraitRef {
+    /// Converts an `ast::PathType` to a `hir::TraitRef`.
+    pub(crate) fn from_ast(ctx: &LowerCtx, node: ast::PathType) -> Self {
+        // FIXME: Use `Path::from_src`
+        TraitRef { path: node.path().and_then(|it| ctx.lower_path(it)).unwrap(/*FIXME*/) }
```

Comment:
I'm unsure how to handle the fallible case here, have to check the call sites again whether returning an Option is correct here? TypeRef just has an error variant here

Reply:
Maybe having this be an enum `Path(Path), Error` makes sense just because of symmetry with `TypeRef`, instead of using an option here?

Reply:
Ye, I think thats probably the best option

Reply:
I guess the question is whether we're fine treating `impl {syntax error} for Type` the same as `impl Type`... and we probably are :thinking: